pkg/pipeline/encode/opentelemetry/encode_otlpmetrics.go

```diff
@@ -0,0 +1,300 @@
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"k8s.io/utils/strings/slices"
+)
+
+// TODO: Refactor the code that is common with encode_prom
+const (
+	defaultExpiryTime = time.Duration(2 * time.Minute)
+	flpMeterName      = "flp_meter"
+)
+
+type counterInfo struct {
+	counter *metric.Float64Counter
+	info    api.PromMetricsItem
+}
+
+type gaugeInfo struct {
+	gauge *metric.Float64ObservableGauge
+	info  api.PromMetricsItem
+	obs   Float64Gauge
+}
+
+// TBD: Handle histograms
+/*
+type histoInfo struct {
+	histo *metric.Float64Histogram
+	info  api.PromMetricsItem
+}
+*/
+
+type EncodeOtlpMetrics struct {
+	cfg      api.EncodeOtlpMetrics
+	ctx      context.Context
+	res      *resource.Resource
+	mp       *sdkmetric.MeterProvider
+	counters []counterInfo
+	gauges   []gaugeInfo
+	//histos           []histoInfo
+	//aggHistos        []histoInfo
+	expiryTime       time.Duration
+	mCache           *putils.TimedCache
+	exitChan         <-chan struct{}
+	meter            metric.Meter
+	metricsProcessed prometheus.Counter
+	metricsDropped   prometheus.Counter
+}
+
+// Encode encodes a metric to be exported
+func (e *EncodeOtlpMetrics) Encode(metricRecord config.GenericMap) {
+	log.Tracef("entering EncodeOtlpMetrics. entry = %v", metricRecord)
+
+	// Process counters
+	for _, mInfo := range e.counters {
+		labels, value, _ := e.prepareMetric(metricRecord, &mInfo.info)
+		if labels == nil {
+			continue
+		}
+		// set attributes using the labels
+		attributes := obtainAttributesFromLabels(labels)
+		(*mInfo.counter).Add(e.ctx, value, metric.WithAttributes(attributes...))
+		e.metricsProcessed.Inc()
+	}
+
+	// Process gauges
+	for _, mInfo := range e.gauges {
+		labels, value, key := e.prepareMetric(metricRecord, &mInfo.info)
+		if labels == nil {
+			continue
+		}
+		// set attributes using the labels
+		attributes := obtainAttributesFromLabels(labels)
+		mInfo.obs.Set(key, value, attributes)
+		e.metricsProcessed.Inc()
+	}
+	// TBD: Process histograms
+}
+
+func (e *EncodeOtlpMetrics) prepareMetric(flow config.GenericMap, info *api.PromMetricsItem) (map[string]string, float64, string) {
+	val := e.extractGenericValue(flow, info)
+	if val == nil {
+		return nil, 0, ""
+	}
+	floatVal, err := utils.ConvertToFloat64(val)
+	if err != nil {
+		return nil, 0, ""
+	}
+
+	entryLabels, key := encode.ExtractLabelsAndKey(flow, info)
+	// Update entry for expiry mechanism (the entry itself is its own cleanup function)
+	_, ok := e.mCache.UpdateCacheEntry(key, entryLabels)
+	if !ok {
+		e.metricsDropped.Inc()
+		return nil, 0, ""
+	}
+	return entryLabels, floatVal, key
+}
+
+func (e *EncodeOtlpMetrics) extractGenericValue(flow config.GenericMap, info *api.PromMetricsItem) interface{} {
+	for _, filter := range info.GetFilters() {
+		val, found := flow[filter.Key]
+		switch filter.Value {
+		case "nil":
+			if found {
+				return nil
+			}
+		case "!nil":
+			if !found {
+				return nil
+			}
+		default:
```

Comment:
The code that came from encode_prom is obsolete, see https://github.com/netobserv/flowlogs-pipeline/pull/513/files#diff-30d11303a4ad1ec82e4f39e3ae6dc7d044c8dd370c536a98ba90febbfe2d425aR209

Reply:
basically: these special keywords "nil" / "!nil" have been replaced with a more explicit API where you can have more advanced filters, e.g. using regexp